backend/main/src/Services/CreateNewUserService/CreateANewUserService.ts

```diff
@@ -0,0 +1,81 @@
+import { CreateUserController } from "src/User/Controllers/CreateUserController";
+import { CreateContainerController } from "src/Group/Controllers/CreateContainerController";
+import { CreateGroupController } from "src/Group/Controllers/CreateGroupController";
+import {
+	BadRequestStatus,
+	Controller,
+	ForbiddenStatus,
+	InternalServerErrorStatus,
+} from "src/Shared";
+import { NishikiDynamoDBClient } from "src/Shared/Adapters/DB/NishikiTableClient";
+
+interface ICreateANewUser {
+	emailAddress: string;
+	name?: string;
+}
+
+/**
+ * Create a new user, container, and group.
+ */
+export class CreateANewUserService extends Controller<ICreateANewUser> {
+	constructor(
+		readonly createUser: CreateUserController,
+		readonly createGroup: CreateGroupController,
+		readonly createContainer: CreateContainerController,
+		readonly nishikiDynamoDBClient: NishikiDynamoDBClient,
+	) {
+		super();
+	}
+
+	async handler(input: ICreateANewUser) {
+		// create a new user
+		const createUserResult = await this.createUser.execute(input);
+		if (!(createUserResult.status === "CREATED" && createUserResult.body)) {
+			return createUserResult as BadRequestStatus | InternalServerErrorStatus;
+		}
+		const userData = createUserResult.body;
+
+		const userId = userData.id;
+		const userName = userData.name;
+
+		// crate a new group
+		const createGroupResult = await this.createGroup.execute({
+			name: `${userName}'s group`,
+			userId: userId,
+		});
+
+		// when the creating a group failed, delete the user for rollback.
+		if (!(createGroupResult.status === "CREATED" && createGroupResult.body)) {
+			await this.nishikiDynamoDBClient.deleteUser(userId);
+			// TODO: delete group
+			return createGroupResult as BadRequestStatus | InternalServerErrorStatus;
+		}
+
+		const groupData = createGroupResult.body;
+
+		const groupId = groupData.id;
+
+		// create a new container
+		const createContainerResult = await this.createContainer.execute({
+			userId,
+			name: `${userName}'s container`,
+			groupId,
+		});
+
+		if (
+			!(createContainerResult.status === "CREATED" && createUserResult.body)
+		) {
+			// TODO: delete user and group.
+			await Promise.all([
+				this.nishikiDynamoDBClient.deleteUser(userId),
+				this.nishikiDynamoDBClient.deleteGroup(groupId),
+			]);
```

Comment:
I'm not sure, how we delete the resources. You know, the detection of each resource is complex, since we are using DynamoDB. We need to get an agreement about how to delete these resources, delete them inside DyanamoClient, create a new service, and so on...

Reply:
Thank you for considering it. Let me confirm what you mean by `these resources`. Does it mean the Group and Containers belonging to it. I thought only deleting Group was complicated because we can't delete all belonging containers at once. IMO, it's better to write a deleting container operation in a delete method of the GroupRepository cuz the implementation depends on the DB system.
I may overlook some other cases. If there are other cases, let me know.

Reply:
Thank you for your reply.
I was worried about the duplicate procedure of deletion. When the user is deleted, the group that has only its user should be deleted. When the group is deleted, the relation with user data also must be deleted. So, I thought the deletion of a user is done by the service.
But now I found that I don't need to care about the deletion operation, since the group deleting is done in the group repository. So, I can fulfill to implement this service using the repositories. Thank you.